### PR TITLE
[Java] Add entry to run custom test using bazel

### PR DESCRIPTION
**Pull request by lixin-wei**
## Why are these changes needed?

Now we can run custom java tests by:

0. `cp testng_custom_template.xml testng_custom.xml`
1. Specify test class/method in `testng_custom.xml`
2. `bazel test //java:custom_test --test_output=streamed`

### CODE REVIEW

**Review by jovany-wang — APPROVED** [review]
Nice change!

File: java/BUILD.bazel [hunk 1]
```diff
@@ -177,11 +177,37 @@ define_java_module(
     ],
 )
 
-java_binary(
+java_test(
     name = "all_tests",
     args = ["java/testng.xml"],
-    data = ["testng.xml"],
+    data = [
+        "testng_custom.xml",
```
Comment (jovany-wang) [inline java/BUILD.bazel:184]: It seems we shouldn't change this line.

**Review by jovany-wang — COMMENTED** [review]
We only need to change `java_binary` as `java_test` for `all_tests`. And other changes should be checked in `custom_test/`

File: java/BUILD.bazel (hunk 1 above)
Comment (lixin-wei) [reply java/BUILD.bazel:184]: Nice catch! Fixed!

**Review by jovany-wang — APPROVED** [review]
LGTM

**Comment by lixin-wei** [comment]
@jovany-wang Failed tests seem unrelated, would you please merge this PR?

**Comment by jovany-wang** [comment]
> @jovany-wang Failed tests seem unrelated, would you please merge this PR?

It seems some tests has been pending for a long time. @simon-mo Do you know the reason?

**Pull request merged**